server/api/runtime_handlers/daskjob.py

```diff
@@ -321,7 +321,20 @@ def enrich_dask_cluster(
         or "daskdev/dask:latest"
     )
     env = spec.env
-    env.extend(function.generate_runtime_k8s_env())
+    for i, runtime_env in enumerate(function.generate_runtime_k8s_env()):
+        found = False
+        for j, function_env in enumerate(env):
+            if runtime_env["name"] == function_env["name"]:
+                # gc
+                del env[j]
+
+                # prioritize the function runtime env var over the given one
+                env[j] = function_env
+                found = True
+                break
+        if not found:
+            env.append(runtime_env)
+
```

Comment:
I suggest changing this code to something a bit simpler with less deletes and breaks.
I will say however the code comment `# prioritize the function runtime env var over the given one` doesn't really describe what's happening, you called the spec.env function env and you are prioritizing that spec.env over the runtime env generated in k8s. So my suggestion has the same logic but I called `function_env` -> `spec_env`. If your comment meant we should prioritize the runtime env, then we can filter the spec env instead and then add the runtime env.
```suggestion
    def _not_in_spec_env(_runtime_env):
        for _env in env:
            if _runtime_env["name"] == _env["name"]:
                return False
        return True
    
    runtime_env = filter(function.generate_runtime_k8s_env(), _not_in_spec_env)
    env.extend(runtime_env)
```